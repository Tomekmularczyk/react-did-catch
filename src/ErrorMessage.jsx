import React from 'react'
import Types from 'prop-types'
import browserInfo from './browserInfo'
import defaultStyles, { mergeStyles } from './styles'
import Collapsible from './Collapsible'

class ErrorMessage extends React.Component {
  state = {
    isExpanded: false
  }

  toggleIsExpanded = () => this.setState(prev => ({ isExpanded: !prev.isExpanded }))

  render () {
    const { isExpanded } = this.state
    const { error, info } = this.props

    const styles = mergeStyles(defaultStyles, this.props.customStyles)
    return (
      <div style={styles.container}>
        <h5 style={styles.errorMessage} onClick={this.toggleIsExpanded}>
          {error.toString()}
          <span style={styles.browserInfo}>({`${browserInfo.name} ${browserInfo.version}`})</span>
          <div style={isExpanded ? { ...styles.arrow, transform: 'rotate(180deg)' } : styles.arrow} />
        </h5>
        <Collapsible isExpanded={isExpanded}>
          <div style={styles.componentStack}>
            {info.componentStack.trim()}
          </div>
        </Collapsible>
      </div>
    )
  }
}

ErrorMessage.propTypes = {
  error: Types.object.isRequired,
  info: Types.shape({
    componentStack: Types.string.isRequired
  }).isRequired,
  customStyles: Types.shape({
    container: Types.object,
    errorMessage: Types.object,
    componentStack: Types.object,
    browserInfo: Types.object,
    arrow: Types.object
  })
}

export default ErrorMessage
