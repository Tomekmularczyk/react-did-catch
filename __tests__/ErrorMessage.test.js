import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ErrorMessage from '../src/ErrorMessage'

jest.mock('../src/browserInfo', () => ({ name: 'Chrome', version: '60.0.0.1' }))

describe('<ErrorMessage />', () => {
  let component
  beforeEach(() => {
    component = shallow(
      <ErrorMessage
        error={new Error('Error happened')}
        info={{ componentStack: 'Error in\nTestComponent ...' }}
      />
    )
  })

  it('matches snapshots', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
