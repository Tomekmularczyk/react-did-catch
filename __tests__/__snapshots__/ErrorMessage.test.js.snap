// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> matches snapshots 1`] = `
<div
  style={
    Object {
      "color": "black",
      "fontSize": "1rem",
      "margin": "auto",
      "textAlign": "left",
      "whiteSpace": "pre-line",
    }
  }
>
  <h5
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
        "margin": "0",
      }
    }
  >
    Error: Error happened
    <span
      style={
        Object {
          "fontWeight": "normal",
        }
      }
    >
      (
      Chrome 60.0.0.1
      )
    </span>
    <div
      style={
        Object {
          "borderColor": "transparent transparent black transparent",
          "borderStyle": "solid",
          "borderWidth": "0 5px 5px 5px",
          "display": "inline-block",
          "height": "0",
          "margin": "auto auto auto 5px",
          "transition": "transform .3s ease",
          "verticalAlign": "middle",
          "width": "0",
        }
      }
    />
  </h5>
  <Collapsible
    isExpanded={false}
  >
    <div
      style={
        Object {
          "fontSize": ".8rem",
          "margin": "0",
          "overflow": "scroll",
        }
      }
    >
      Error in
TestComponent ...
    </div>
  </Collapsible>
</div>
`;
